### DISCUSSION TITLE
Jest unit test error from mockAllIsIntersecting since installing intersection-observer

### DISCUSSION THREAD

**Discussion by Muzyar-rad** [thebuilder/react-intersection-observer#571] [Q&A]
Hi,

Do you know why I am getting this following error since installing `npm i intersection-observer`?

This is the error I receive when using **mockAllIsIntersecting** in my test:

`TypeError: Cannot read property 'isIntersecting' of undefined`

**Comment by thebuilder** [comment]
Can you post the full stack trace? Installing `intersection-observer` shouldn't have an impact. It's unsupported in Jest (and JSDom), so it should be mocked. The `test-utils.js` should do it automatically.

**Reply by Muzyar-rad** [reply]
yes. my test was working before. but since `npm i intersection-observer`,
The test which is the following:

```
import { mockAllIsIntersecting} from 'react-intersection-observer/test-utils';

  it('render the Dealer widget', () => {
    render(
      <MockedProvider>
        <DealerCard {...props} />
      </MockedProvider>
    );
    mockAllIsIntersecting(true); // element is in view and renders iframe element
    const container = document.querySelector('iframe');
    expect(container).toBeInTheDocument();
  });
```
fails with the following error:
**TypeError: Cannot read property 'isIntersecting' of undefined**

**Reply by Muzyar-rad** [reply]
I am using:

```
"react-intersection-observer": "^9.2.0",
 "intersection-observer": "^0.12.2"
```

**Reply by thebuilder** [reply]
Are you importing `intersection-observer` anywhere? It might break, since it skips mocking intersection observer if `window.IntersectionObserver` is defined.

**Reply by Muzyar-rad** [reply]
yes I am importing that actually. Since I am importing a parent package that in fact that package is doing `-import intersection-observer`.
What is my alternative? Can I still use your package with the line of code mentioned above?

**Reply by thebuilder** [reply]
You could try running `setupIntersectionMocking(jest.fn)`  in your test. 
This is normally run in `beforeEach`, and it replaces `window.IntersectionObserver` with a mocked version. 

Maybe try logging out  `window.IntersectionObserver` first, to determine if it is in fact a mocked version?

**Reply by Muzyar-rad** [reply]
yes you are right. `setupIntersectionMocking(jest.fn)` in `beforeEach` worked and it seems to start mocking the intersectionObserver. Now I am getting this error though:
`TypeError: item.callback is not a function`
